automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/NotebookInstallSpec.scala

```diff
@@ -0,0 +1,210 @@
+            |status_code(r)
+          """.stripMargin
+
+        notebookPage.executeCell(httpGetTest) shouldBe Some("200")
+      }
+    }
+
+    // See https://github.com/DataBiosphere/leonardo/issues/398
+    "should be able to install mlr" in withWebDriver { implicit driver =>
+      Orchestration.billing.addUserToBillingProject(billingProject.value, ronEmail, Orchestration.billing.BillingProjectRole.User)(hermioneAuthToken)
+
+      withNewNotebook(ronCluster, RKernel) { notebookPage =>
+        // mlr: machine learning in R
+        // https://github.com/mlr-org/mlr
+
+        // it may take a little while to install
+        val installTimeout = 5.minutes
+
+        val installOutput = notebookPage.executeCell("""devtools::install_github("mlr-org/mlr")""", installTimeout)
+        installOutput shouldBe 'defined
+        installOutput.get should include ("Installing mlr")
+        installOutput.get should not include ("Installation failed")
+
+        // Make sure it was installed correctly; if not, this will return an error
+        notebookPage.executeCell("library(mlr)") shouldBe Some("Loading required package: ParamHelpers")
+        notebookPage.executeCell(""""mlr" %in% installed.packages()""") shouldBe Some("TRUE")
+      }
+    }
+
+    //Test to check if extensions are installed correctly
+    //Using nbtranslate extension from here:
+    //https://github.com/ipython-contrib/jupyter_contrib_nbextensions/tree/master/src/jupyter_contrib_nbextensions/nbextensions/nbTranslate
+    "should install user specified notebook extensions" in withWebDriver { implicit driver =>
+      val clusterName = ClusterName("user-jupyter-ext" + makeRandomId())
+      withNewCluster(billingProject, clusterName, ClusterRequest(Map(), Option(testJupyterExtensionUri), None)) { cluster =>
+        withNewNotebook(cluster) { notebookPage =>
+          notebookPage.executeCell("1 + 1") shouldBe Some("2")
+          //Check if the mark up was translated correctly
+          notebookPage.translateMarkup("Hello") should include("Bonjour")
+        }
+      }
+    }
+
+    val sparkCommandToFail =
+      """try:
+        |    name = sc.appName
+        |
+        |except NameError as err:
+        |    print(err)""".stripMargin
+
+    Seq(Python2, Python3).foreach { kernel =>
+      s"should be able to pip install packages using ${kernel.string}" in withWebDriver { implicit driver =>
+        withNewNotebook(ronCluster, kernel) { notebookPage =>
+          // install tensorflow
+          pipInstall(notebookPage, kernel, "tensorflow")
+          notebookPage.saveAndCheckpoint()
+        }
+
+        // need to restart the kernel for the install to take effect
+        withNewNotebook(ronCluster, kernel) { notebookPage =>
+          // verify that tensorflow is installed
+          verifyTensorFlow(notebookPage, kernel)
+        }
+      }
+
+      s"should NOT be able to run Spark using ${kernel.string}" in withWebDriver { implicit driver =>
```

Comment:
I wonder if this test would make more sense in `NotebookHailSpec`, since it's verifying Spark and not installing packages.